src/Features/Core/Portable/Completion/Providers/TypeImportCompletionService.cs

```diff
@@ -251,6 +252,7 @@ static string GetReferenceKey(PortableExecutableReference reference)
                             // If any of the generic overloads is public, then the completion item is considered public.
                             var item = TypeImportCompletionItem.Create(overloadInfo.BestGenericOverload, containingNamespace);
                             var isPublic = overloadInfo.ContainsPublicGenericOverload;
+                            item.IsCached = true;
```

Comment:
TypeImportCompletionItem.Create should just set this value right?

Reply:
It can. But I wanted to keep this close to where the cache is.

Reply:
Ok.  Though it seems potentially risky.  If someone constructs this elsewhere, they'd need to know to do this.  Up to you!

Reply:
If someone calling `TypeImportCompletionItem.Create` without using the cached service, then they shouldn't set this unless they implement their own cache. I think this should never happen.